docs/spec_draft.md

```diff
@@ -663,6 +664,142 @@ produces an implementation-defined value.
+3. `id = nd` where `d` $\notin$ `rhs_contracting_dimensions` $\cup$ `rhs_batching_dimensions`
+
+The summation is from `0` to prod(dim(`lhs_contracting_dimensions`, `i`)) for all `i` in [0, size(`lhs_contracting_dimensions`))
+for all `mk`, `nk` combinations defined as:
+* `mk` in [0, dim(`lhs`, `k`)) for all `k` $\in$ `lhs_contracting_dimensions`
+* `nk` in [0, dim(`rhs`, `k`)) for all `k` $\in$ `rhs_contracting_dimensions`
+
+`precision_config` controls the tradeoff between speed and accuracy for
+computations on accelerator backends. This can be one of the following:
+
+  * `DEFAULT`: Fastest mode, but least accurate.
+  * `HIGH`: Slower, but more accurate.
+  * `HIGHEST`: Slowest, but most accurate.
+
+### Inputs
+
+| Name                         | Type                                         |
+|------------------------------|----------------------------------------------|
+| `lhs`                        | tensor of any supported types                |
+| `rhs`                        | tensor of any supported types                |
+| `lhs_batching_dimensions`    | 1-dimensional tensor constant of type `si64` |
+| `rhs_batching_dimensions`    | 1-dimensional tensor constant of type `si64` |
+| `lhs_contracting_dimensions` | 1-dimensional tensor constant of type `si64` |
+| `rhs_contracting_dimensions` | 1-dimensional tensor constant of type `si64` |
+| `precision_config`           | constant of type `si64`                      |
+
+### Outputs
+
+| Name     | Type                          |
+|----------|-------------------------------|
+| `result` | tensor of any supported types |
+
+### Constraints
+
+  * (C1) `lhs` and `rhs` have the same element type.
+  * (C2) size(`lhs_batching_dimensions`) $=$ size(`rhs_batching_dimensions`)
+  and size(`lhs_contracting_dimensions`) $=$ size(`rhs_contracting_dimensions`).
+  * (C3) `lhs_batching_dimensions` and `lhs_contracting_dimensions` combined are
+  unique.
+  * (C4) `rhs_batching_dimensions` and `rhs_contracting_dimensions` combined are
+  unique.
+  * (C5) 0 $\le$ `lhs_batching_dimensions[i]` $\lt$ rank(`lhs`) for all `i`
+  $\in$ [0, size(`lhs_batching_dimensions`)).
+  * (C6) 0 $\le$ `lhs_contracting_dimensions[i]` $\lt$ rank(`lhs`) for all `i`
+  $\in$ [0, size(`lhs_contracting_dimensions`)).
+  * (C7) 0 $\le$ `rhs_batching_dimensions[d]` $\lt$ rank(`rhs`) for all `i`
+  $\in$ [0, size(`rhs_batching_dimensions`)).
+  * (C8) 0 $\le$ `rhs_contracting_dimensions[d]` $\lt$ rank(`rhs`) for all `i`
+  $\in$ [0, size(`rhs_contracting_dimensions`)).
+  * (C9) `dim(lhs, lhs_batching_dimensions[i])` =
+  `dim(rhs, rhs_batching_dimensions[i])` for all `i` $\in$ [0,
+  size(`lhs_batching_dimensions`)).
+  * (C10) `dim(lhs, lhs_contracting_dimensions[i])` =
+  `dim(rhs, rhs_contracting_dimensions[i])` for all `i` $\in$ [0,
+  size(`lhs_contracting_dimensions`)).
+  * (C11) `dim(result, i)` is equal to (in order):
```

Comment:
The i in `result` is different than  index `i` of lhs and rhs

How about:

The shape of `result` is given by:

{`lhs_batching_dimensions[i]` :  for all 0 $\le$ `i` $\lt$ size(`lhs_batching_dimensions`)} + {`dim(lhs, i)` : for all i $\in$ [0, rank(`lhs`)) where `i` $\notin$ `lhs_batching_dimensions` and `i` $\notin$ `lhs_contracting_dimensions`} + {`dim(rhs, i)` : for all i $\in$ [0, rank(`rhs`)) where `i` $\notin$ `rhs_batching_dimensions` and `i` $\notin$ `rhs_contracting_dimensions`}

Reply:
Thanks, this is much cleaner